t/lib/WTSI/NPG/HTS/TreePublisherTest.pm

```diff
@@ -16,15 +16,44 @@ use WTSI::NPG::iRODS;
 
 use JSON;
 use Readonly;
+use IPC::System::Simple qw(system);
 
 Log::Log4perl::init('./etc/log4perl_tests.conf');
 
 my $pid          = $PID;
 my $test_counter = 0;
 my $data_path    = 't/data';
+my $bin_path     = 'bin';
 
 my $irods_tmp_coll;
 
+sub write_json {
+  my ($json_filename, $irods_collection) = @_;
+  if (defined $json_filename && (not $json_filename =~ qr/^\s*$/sxm)
+      && defined $irods_collection && (not $irods_collection =~ qr/^\s*$/sxm)) {
+    Readonly::Scalar my $JSON_FILE_VERSION => '1.0';
+
+    my ($json_fh, $json_hash);
+    open $json_fh, '>:encoding(UTF-8)', $json_filename or
+      self->logcroak(q[could not open ml warehouse json file] .
+      qq[$json_filename]);
+    $json_hash = {
+      version  => $JSON_FILE_VERSION,
+      irods_collection => $irods_collection
+    };
+    print $json_fh encode_json($json_hash) or
+      self->logcroak(q[could not write to ml warehouse json file ] .
+      qq[$json_filename]);
+
+    close $json_fh or
+      self->logcroak(q[could not close ml warehouse json file] .
+      qq[$json_filename]);
+  } else {
+    self->logcroak(q[Wrong parameters in write_json]);
+  }
+  return 1;
+}
```

Comment:
This seems to be a copy of the code you added to the script. What's your aim here?

Reply:
Testing a literal copy of some code is not the same as testing the original code! Can you see they they can diverge? What happens if someone changes the code in `npg_publish_tree.pl` in such a way as it produces an incorrect result? This test would still pass.

Reply:
Yes, that's right. Indeed, it doesn't make sense to test a copy. It would be easier in this case if the write_json function was a class method of TreePublisher.